amity/room.py

```diff
@@ -1,24 +1,82 @@
+from random import choice
+from person import Person
+
+
 class Room(object):
     ''' Room Class
 
         This is the Super Class that handles the creation of Rooms
         in the Amity System. It relies heavily on the information it
         gets from it's Child Classes.
     '''
+    number_of_offices = 0
+    number_of_living_spaces = 0
+    total_rooms = 0
+    rooms = {"Offices": {}, "Living Spaces": {}}
+    error = ""
 
-    def __init__(self, name, type):
+    def __init__(self, name, room_type, capacity):
+        room_key = room_type + "s"
+        room_id = name.lower()
+        all_room_ids = Room.rooms['Offices'].keys() + Room.rooms['Living Spaces'].keys()
+        if name.lower() in all_room_ids:
+            Room.error = "A room named '" + name + "' already exists. Please choose another name."
+        else:
+            self.rooms[room_key][room_id] = {}
+            self.rooms[room_key][room_id]["Room Name"] = name
+            self.rooms[room_key][room_id]["Room ID"] = room_id
+            self.rooms[room_key][room_id]["Capacity"] = capacity
+            self.rooms[room_key][room_id]["Total Persons"] = 0
+            self.rooms[room_key][room_id]["Occupants"] = []
+            if room_type == "Office":
```

Comment:
For this, at no point during the execution of the application does the user have access to this class directly. The Room class receives the ```room_type``` from it's child classes and therefore, this variable can only either be ```Office``` or  ```Living Space```.